File: sdk/client/java/openapi-test/src/test/java/opensdk/sdk/apis/debug/runtimeDebugging/DebugFreeOSMemoryTest.java
package opensdk.sdk.apis.debug.runtimeDebugging;

import opensdk.sdk.apis.constant.UrlConstants;
import opensdk.sdk.models.DebugFreeOSMemoryResponse;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.klaytn.OpenSDK;

import java.io.IOException;

import static org.junit.jupiter.api.Assertions.assertNotNull;
import static org.junit.jupiter.api.Assertions.assertNull;

@DisplayName("Debug RPC Test")
public class DebugFreeOSMemoryTest {
    private final OpenSDK sdk = new OpenSDK(UrlConstants.SERVER_URL);

    @Test
    @DisplayName("RPC debug_freeOSMemory")
    void whenRequestValid_ThenCall200ResponseReturns() throws IOException {
        DebugFreeOSMemoryResponse response = sdk.debug.freeOSMemory().send();

        assertNotNull(response);
        assertNull(response.getError());
    }
}

File: web3rpc/sdk/client/javascript/openapi-test/test/debug/runtimeDebugging/DebugFreeOSMemory.test.js
const OpenSdk = require("opensdk-javascript");
const { expect } = require("@jest/globals");
const { RPC } = require("../../constant");

const sdk = new OpenSdk(new OpenSdk.ApiClient(RPC));

describe('debug_freeOSMemory API', () => {
    test('should return debug_freeOSMemory', (done) => {

        let callbackOne = function (error, data, response) {
            expect(error).toBeNull();
            expect(data).toBeDefined();
            expect(data).toBe(null);
            done();
        };

        sdk.debug.freeOSMemory({}, callbackOne);
    });
});

